### ISSUE TITLE
Unexpected Cuda(0) Usage

### ISSUE THREAD

**Issue by neveranever98** [LaurentMazare/tch-rs#463]
I was using a machine with eight graphics cards to train a model. However, I didn't know how to use 2 or more GPU at the same time to train a single model (this is another problem). So I chose cuda(3) in my code to train a model, but after the bold line of code in the figure below, which forward function was called, cuda(0) seemed being used. I don't know why.

<img width="660" alt="image" src="https://user-images.githubusercontent.com/97876140/158318658-bd494876-b290-4414-9c43-2a283e15bc37.png">

And this is the forward function, No CUDA(0) is explicitly called. 

<img width="855" alt="image" src="https://user-images.githubusercontent.com/97876140/158319510-2a745890-efcd-454e-839d-250e14bfc637.png">

In conclusion, I actually have two questions.
1. why cuda(0) is called in net.forward() function
2. how to use 2 GPU to train one model at the same time to speed up.

**Comment by LaurentMazare** [comment]
I'm not sure how the C++ library converts the gpu device ids to what appears in nvidia-smi so cannot really comment here. Re multi-gpu training, did you try using both `cuda(1)` and `cuda(2)` in the same code to see if two gpus get used?
